Add Portlibrary Filestream API

youngar opened the pull request.

alin00 commented on port/common/omrfilestream.c:
```diff
@@ -0,0 +1,488 @@
+static int32_t
+findError (int32_t errorCode)
+{
+	switch (errorCode)
```
Please follow coding standard for formatting switch statements: https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#formatting-switch-statements

alin00 commented on fvtest/porttest/omrfilestreamTest.cpp:
```diff
@@ -0,0 +1,2692 @@
+
+	rc = omrfilestream_sync(writeFile);
+	if (0 > rc) {
+		outputErrorMessage(PORTTEST_ERR
```
should be uint16_t here and line 1914 below

alin00 commented on fvtest/porttest/omrfilestreamTest.cpp:
```diff
@@ -0,0 +1,2692 @@
+	const char *filename = "omrfilestream_test_buffering.tst";
+	const char output[]  = "testing\noutput";
+	char input[]         = "xxxxxxxxxxxxxx";
+	I_32 characterCount = sizeof(output);
```
use int32_t instead of I_32

charliegracie commented on port/common/omrfilestream.c:
```diff
@@ -0,0 +1,488 @@
+	/* Get the native file descriptor */
+	fileDescriptor = portLibrary->file_convert_omrfile_fd_to_native_fd(portLibrary, fd);
+#if defined(J9ZOS390)
+	switch (fileDescriptor) {
```
What is the rationale for the empty swtich?

youngar commented:
@alin00 I un-indented the switch statement and replaced all the usages of I_* types in omrfilestreamTest.cpp (the other files were fine).

@charliegracie Its a mistake, the logic for that switch was moved into file_convert_omrfile_fd_to_native_fd and file_convet_native_fd_to_omr_fd.  I have deleted the empty switch.

charliegracie commented on port/unix/omrfile.c:
```diff
@@ -1237,11 +1237,31 @@ intptr_t
+#if (FD_BIAS != 0)
+	/* Do NOT add the FD_BIAS to standard streams */
+	switch (nativeFD) {
+		case OMRPORT_TTY_IN:
```
This switch is formatted incorrectly.  Can you fix it please?

charliegracie commented on port/win32/omrfile.c:
```diff
@@ -58,6 +58,26 @@ toHandle(OMRPortLibrary *portLibrary, intptr_t fd)
+static intptr_t
+fromHandle(OMRPortLibrary *portLibrary, HANDLE handle)
+{
+	if (handle == PPG_tty_consoleInputHd)
```
Can you add the "{}" brackets for the if statements please?

charliegracie commented on port/win32/omrfilestream.c:
```diff
@@ -0,0 +1,424 @@
+{
+	switch (errorCode)
+	{
+		case EACCES:
```
Can you fix the formatting please? Including any fall through cases.

youngar commented:
@charliegracie Updated the commit with your suggestions
